db/schema.rb

```diff
@@ -401,6 +401,7 @@
     t.boolean "race_refused_to_answer"
     t.boolean "latest_assessment_symptomatic", default: false
     t.date "first_positive_lab_at"
+    t.boolean "no_symptom_history", default: false
```

Comment:
After chewing on this for a while, I'm not convinced we should add a field to the patient model for this. Once you get through enrollment, it's not something a user can modify directly, and it seems it's essentially just a computed field, based entirely on assessments and/or symptom onset date. Storing it as a separate field opens up the possibility for bugs where `no_symptom_history` could get out of sync with what it's "supposed" to be based on those other criteria. For example, the situation I called out in another comment about setting symptom onset after enrollment. Thoughts? Am I missing something?

Reply:
So I actually had this implemented without a `no_symptom_history` field but then the following requirement was specifically added to the [ticket](https://tracker.codev.mitre.org/browse/SARAALERT-562) description:
- This boolean NO SYMPTOMS REPORTED should be a new field on the Patient record and it should be exportable everywhere that continuous exposure already is

I do see how this field can seem somewhat redundant and can have the potential to end up in an inconsistent state with the data, which is something we definitely need to be careful about. However, if the PO team wants this to be something that's exported, I think it would make sense for this to be its own field to avoid having to compute this value for every patient during the time of export. I also think that if we ever want to perform queries based off this field (which I think seems likely for a future advanced filter option), it will be much more performant to be able to query based on this field instead of having to compute it.

Reply:
Hey so I considered this some more and chatted with @holmesie on it briefly. We think this field should not be stored in the database. It just unnecessarily opens up the potential for bugs, and since it is a derived field it should be implemented as a derived field.

After thinking more about the performance implications, I'm not seeing how it would be performance hit. Isn't the value just always going to be `!patient.symptom_onset.present?`.

Reply:
That makes sense, is this something we should verify with the PO team before removing it since they specifically asked for it to be a new field? Or does that not really matter because from a UI perspective, they won't really notice a change?

Reply:
done here: https://github.com/SaraAlert/SaraAlert/pull/747/commits/15a6d7a0c71fee53608f709f7c55fb913842dbd8#diff-cbf8b25d6d853acf58fa9057841f407d29ffe90649c75cf9e3f51b2d6e3aa1d3L404

Reply:
Yeah exactly my thinking- it's all the same from the user perspective, just an implementation detail.